### PR TITLE
Use loops in build flow, support 11 shards

### PR DESCRIPTION
**Pull request by benpatterson**

### CODE REVIEW

**Comment by benpatterson** [comment]
@estute @jzoldak PTAL

**Comment by benpatterson** [comment]
This will be the testeng-ci side. Companion PR in progress [here](https://github.com/edx/edx-platform/pull/14238).

**Comment by estute** [comment]
:+1: Nice. I was unaware of that range syntax

**Comment by cpennington** [comment]
Does it makes sense to do the other jobs w/ shards as well (lms?)

**Comment by benpatterson** [comment]
@cpennington do you mean shorten their length, or improve the `groovy`?

File: jenkins/flow/master/edx-platform-bok-choy-master.groovy [hunk 1]
```diff
@@ -35,42 +35,13 @@ try{
...
-        },
-        {
-          bok_choy_8 = build(subsetJob, sha1: sha1, SHARD: "8", TEST_SUITE: "bok-choy", ENV_VARS: params["ENV_VARS"], PARENT_BUILD: "master #" + build.number, WORKER_LABEL: workerLabel)
-          toolbox.slurpArtifacts(bok_choy_8)
-        },
-        {
-          bok_choy_9 = build(subsetJob, sha1: sha1, SHARD: "9", TEST_SUITE: "bok-choy", ENV_VARS: params["ENV_VARS"], PARENT_BUILD: "master #" + build.number, WORKER_LABEL: workerLabel)
-          toolbox.slurpArtifacts(bok_choy_9)
-        },
+      (1..11).collect { index ->
```
Comment (jzoldak) [inline jenkins/flow/master/edx-platform-bok-choy-master.groovy]: this is fly @benpatterson ! 😍

**Comment by cpennington** [comment]
@benpatterson: I meant improve the ruby.

**Pull request merged**